Resfeber/CoreData/TripsController.swift

```diff
@@ -9,15 +9,20 @@
 import Foundation
 import CoreData
 
-public final class TripService {
+public final class TripsController {
     // MARK: - Properties
-    let context = CoreDataStack.shared.mainContext
+    let context: NSManagedObjectContext
+    let coreDataStack: CoreDataStack
     
-    private(set) var searchTrips = [Trip]()
+    init(managedObjectContext: NSManagedObjectContext = CoreDataStack.shared.mainContext,
+         coreDataStack: CoreDataStack = CoreDataStack.shared) {
```

Comment:
You shouldn't need to initialize with a `CoreDataStack` because everything should just be using your `shared` instance

Reply:
This is set up so the unit tests can create a mock CoreDataStack class and inject it into the TripsController class.